x-pack/plugins/cases/server/client/attachments/get.ts

```diff
@@ -314,3 +324,45 @@ export async function getAll(
     });
   }
 }
+
+/**
+ * Retrieves a count of all alerts attached to a specific case.
+ *
+ * @ignore
+ */
+export const countAlertsAttachedToCase = async (
+  caseId: string,
+  clientArgs: CasesClientArgs,
+  casesClient: CasesClient
+): Promise<AlertsCount> => {
+  const { unsecuredSavedObjectsClient, authorization, attachmentService, logger } = clientArgs;
+
+  try {
+    // This will perform an authorization check to ensure the user has access to the parent case
+    const theCase = await casesClient.cases.get({
+      id: caseId,
+      includeComments: false,
+      includeSubCaseComments: false,
+    });
+
+    const { filter: authorizationFilter } = await authorization.getAuthorizationFilter(
+      Operations.countAlertsAttachedToCase
+    );
+
+    const alertsCount = await attachmentService.countAlertsAttachedToCase({
+      unsecuredSavedObjectsClient,
+      caseId: theCase.id,
+      filter: authorizationFilter,
+    });
+
+    return AlertsCountRt.encode({
```

Comment:
Side note, since the service doesn't return full saved objects (only a count) we have nothing to pass to `ensureSavedObjectsAreAuthorized` so I'm not calling it for this handler.

The cardinality aggregation specifically doesn't return the ids or owner information that we'd need to call `ensureSavedObjectsAreAuthorized`.